### DISCUSSION TITLE
Data Flow and Clearing House

### DISCUSSION THREAD

**Discussion by musi00** [International-Data-Spaces-Association/DataspaceConnector#66] [Q&A] [answered]
Hello,

I'm trying to set up a demo where I transfer data from provider to consumer and then show that the data flow has been captured by the clearing house. It seems that there isn't a clearing house implementation as of yet.  However, I imagine that the data flow information is still being captured by the connectors themselves.  I have a few questions related to data flow, logging and security of the data:

1) Can I confirm whether there is no clearing house implementation available, if not when is it expected?
2) Where is the data flow information captured in the connectors.  is it just the connector log files under log/dataspaceconnector.log?
3) one of the advertised features of the data space connector is **integrity-protected** logging.  How is the integrity of these logs protected?  Are they somehow signed, if so, where are those signatures stored?
4) Is the data or metadata encypted.  The following screenshot of the metadata in the database table makes me thing that at least the metadata is encrypted:

![image](https://user-images.githubusercontent.com/5386051/109595618-ec43dc80-7ae2-11eb-9928-6478eebbdb01.png)

Thanks in advance for considering my questions.

**Comment by juliapampus** [comment, marked as answer]
Hi, to answer your questions:
1. The Clearing House already exists, but there have been some compatibility issues in the past months. We aim to update the communication in future releases.
2. Currently, we log everything that can be caught via the spring filters. This ends up in the log files and can be retrieved via acutator endpoints (see [Logging](https://github.com/International-Data-Spaces-Association/DataspaceConnector/wiki/Logging)). A specific filter for IDS transactions is currently not available. They are automatically caught along with all http transactions.
3. Audit logging is on our roadmap. We are currently dealing with it theoretically in order to find an appropriate solution. We will then implement this as soon as possible.
4. The metadata and data in the H2 are not encrypted. What you see is binary information. Currently, the Connector itself does not take care of the database encryption.

**Reply by musi00** [reply]
Thanks for your prompt answers.

**Reply by musi00** [reply]
sorry I had one follow up question.  You mentioned that the clearing house already exists but I cannot find a repository for the clearing house implementation, the only hint I found from DSC code is that you have a LogMessageService that refers to a URL that might be the clearing house.  https://ch-ids.aisec.fraunhofer.de/logs/messages.  However this URL is invalid.
